chore: Update contributing.adoc (#2005)
codenvy/che-docs: CONTRIBUTING.adoc

*** link:https://www.nngroup.com/articles/chunking/[Chunking].

** _IBM Style Guide_. 

*** link:https://www.oreilly.com/library/view/the-ibm-style/9780132118989/[The IBM Style Guide, Conventions for Writers and Editors] 

*** link:https://github.com/errata-ai/IBM[A Vale-compatible implementation of IBM's _Developer Editorial Style Guide_]

** Guidelines from the
link:https://redhat-documentation.github.io/modular-docs/[Modular Documentation Initiative].
+
[TIP]
====
For Antora, a link:https://docs.antora.org/antora/2.3/module-directories/#module[_module_] is a collection of content in a component version. This project is referring to Modular Documentation modules as _topics_. To understand the nature of topics, see the link:https://redhat-documentation.github.io/modular-docs/#modular-docs-terms-definitions[Appendix A: Modular Documentation Terms and Definitions].
====

* Use the link:https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow[Git Forking Workflow].

* link:https://www.eclipse.org/che/docs/che-7/hosted-che/[Eclipse Che hosted by Red Hat] is the recommended IDE to edit and builds the content.

> * The relevant tools for all edition, validation and publication workflows are built in the link:https://quay.io/repository/eclipse/che-docs[`che-docs` container].

Reviewer: ```suggestion
* The relevant tools for editing, validation, and publication workflows are built in the link:https://quay.io/repository/eclipse/che-docs[`che-docs` container].
```

.Procedure

. To build the documentation and start the preview server: open _My Workspace / User Runtimes / che-docs / Start preview server_.

> . To navigate the preview, open _Preview_.

Reviewer: ```suggestion
. To navigate to the preview, open _Preview_.
```

> . To validate links: open _User Runtimes / che-docs / Validate links_.

Reviewer: ```suggestion
. To validate the links: open _User Runtimes / che-docs / Validate links_.
```

> . To validate language changes: open _My Workspace / User Runtimes / che-docs / Validate language changes_.

Reviewer: ```suggestion
. To validate the language changes: open _My Workspace / User Runtimes / che-docs / Validate language changes_.
```

> . To validate the language of the files currently open, look at the _Problems_ tab in the _Bottom Panel_. To toggle the view of this tab use the _View > Problems_ menu entry.

Reviewer: ```suggestion
. To validate the language of the currently open files, look at the _Problems_ tab in the _Bottom Panel_. To toggle the view of this tab use the _View > Problems_ menu entry.
```

....
====

== Building and validating Documentation on a local environment using the `che-docs` container

This section describes how to build and validate the documentation on a local environment.

WARNING: This is not the preferred method. For the supported method, see: xref:building-and-validating-the-documentation-using-che[].

.Prerequisites:

* A running installation of link:http://podman.io[podman] or link:http://docker.io[docker].

.Procedure

> . Open a terminal into the project directory, and build the documentation using and instance of the `che-docs` container:

Reviewer: ```suggestion
. Open a terminal into the project directory, and build the documentation using an instance of the `che-docs` container:
```

+
----
$ ./tools/runnerpreview.sh
---- 

. Navigate to `+localhost:4000+` in your browser.

> . Open a second terminal, and attach to the running `che-docs` container. Run all following steps from this context.

Reviewer: ```suggestion
. Open a second terminal, and attach it to the running `che-docs` container. Run all the following commands from this context.
```

+
----
$ podman exec -ti che-docs  bash
----

. To validate links, run:
+
----
$ htmltest
----

. To validate language on changed files, run:
+
----
$ ./tools/validate_language_changes.sh
----

> . To create a new topic using templates, run following command:

Reviewer: ```suggestion
. To create a new topic using the modular templates, run:
```
